improver_tests/precipitation_type/freezing_rain/test_FreezingRain.py

```diff
@@ -195,3 +195,28 @@ def test_no_freezing_threshold(input_cubes):
 
     with pytest.raises(ValueError, match="No 0 Celsius or equivalent threshold"):
         FreezingRain()(input_cubes)
+
+
+def test_realization_matching(
+    precipitation_multi_realization,
+    temperature_multi_realization,
+    expected_probabilities,
+):
+    """Test that only common realizations are used when combining precipitation
+    and temperature cubes. In this case the temperature cube has an additional
+    realization that is not used, with only 2 realizations present in the
+    freezing rain cube."""
+    cubes = iris.cube.CubeList(
+        [*precipitation_multi_realization, temperature_multi_realization]
+    )
+
+    # Sourcing cubes from independently parameterised fixtures, so we will get
+    # a mix of period and instantaneous precip / temperature diagnostics. We
+    # only want to use the matching types.
```

Comment:
This tells me that you should parameterise this test function, not the individual fixtures. I see the same thing at line 180 too. The solution might be to define additional fixtures that define both the precip and temperature cubes for these cases, preventing the mismatched times from occurring which results in useless tests.

Reply:
I've made the various fixtures used in these types of test accept a period argument directly, and then parameterised the tests themselves. In `conftest` I have now got only one fixture that takes params, which is `input_cubes`. There are 8 fewer tests now. The reduction is due to:
- the removal of the tests you've highlighted which were just passed and did nothing when the time coordinates mismatched.
- the simplification of some exception tests that look for too few / many input cubes to only test with one type of input (instantaneous) as the time coordinate is irrelevant to the tests.